main.go

```diff
@@ -69,18 +78,38 @@ func (hc *HealthCheckWebService) apiGetIsPrimary(w http.ResponseWriter, r *http.
 
 func (hc *HealthCheckWebService) apiGetIsReplica(w http.ResponseWriter, r *http.Request) {
 	nodeInfo, err := hc.healthChecker.dataSource.GetNodeInfo()
+
 	if err != nil {
 		// Return a 500. Something bad happened.
 		panic(err)
 	}
 
+	// if byte lag exceeds max_allowable_byte_lag then return 500
+	if hc.maxAllowableByteLagExceeded(r, nodeInfo) {
+		w.WriteHeader(http.StatusServiceUnavailable)
+	}
+
 	if !nodeInfo.IsReplica() {
```

Comment:
this code is reachable after the max allowed byte lag correct?

Reply:
moved this block above the byte lag check